File: TabloidFullStack/TabloidFullStack/client/tabloid/src/components/Post.js
import React from "react";
import { Card, CardImg, CardBody, CardFooter, Button } from "reactstrap";
import { Link, useNavigate } from "react-router-dom";

export const Post = ({ post }) => {
	const navigate = useNavigate();
	return (
		<Card className='m-4'>
			<p className='text-left px-2'>
				Posted by: {post.userProfile.firstName} {post.userProfile.lastName}
			</p>
			<p className='text-left px-2'>
				Author Display Name: {post.userProfile.displayName}
			</p>
			<p className='text-left px-2'>
				Category: {post.categoryId} Published on: {post.publishDateTime}
			</p>
			<CardImg top src={post.imageLocation} alt={post.title} />
			<CardBody>
				<p>
					<Link to={`/post/${post.id}`}>
						<strong>{post.title}</strong>
					</Link>
				</p>
				<p>{post.content}</p>
			</CardBody>
			<CardFooter>
				<Button
					outline
					color='primary'
					onClick={(e) => {
						e.preventDefault();
						navigate(`/Post/${post.id}/Comments`);
					}}
				>
					View Comments
				</Button>
			</CardFooter>
		</Card>
	);
};

File: TabloidFullStack/TabloidFullStack/Models/Post.cs
﻿using System.ComponentModel.DataAnnotations;

namespace TabloidFullStack.Models
{
    public class Post
    {
        [Required]
        public int Id { get; set; }

        [Required]
        public string Title { get; set; }

        [Required]
        public string Content { get; set; }

        public string? ImageLocation { get; set; }

        [Required]
        public DateTime CreateDateTime { get; set; }
        
        public DateTime? PublishDateTime { get; set; }

        [Required] 
        public bool IsApproved { get; set; }


        [Required]
        public int CategoryId { get; set; }
        public Category? Category { get; set; }



        [Required]
        public int UserProfileId { get; set; }

        public UserProfile? UserProfile { get; set; }

    }
}
